aoservicelisting.civix.php

```diff
@@ -175,34 +195,31 @@ public static function endABARelationship($values, $rowNumber, $orgId) {
 
   public static function findDupes($cid, $orgId, &$individualParams, $rel = EMPLOYER_CONTACT_REL, $isOrgOptional = FALSE) {
     if (!empty($cid)) {
-      $currentDetails = civicrm_api3('Contact', 'getsingle', ['id' => $cid]);
-      if ($currentDetails['first_name'] != $individualParams['first_name'] || $currentDetails['last_name'] != $individualParams['last_name']) {
-        $params = ['contact_id_a' => $cid, 'contact_id_b' => $orgId, 'is_active' => 1];
-        $relationships = civicrm_api3('Relationship', 'get', $params);
-        if (!empty($relationships['values'])) {
-          // End Date all relationships as they have either overwritten the data or not.
-          foreach ($relationships['values'] as $relationship) {
-            civicrm_api3('Relationship', 'create', ['id' => $relationship['id'], 'is_active' => 0, 'relationship_type_id' => $relationship['relationship_type_id'], 'end_date' => date('Y-m-d')]);
-          }
-        }
-      } else {
-        $individualParams['contact_id'] = $cid;
+      // Check to see if this contact is a primary contact.
+      if (E::checkPrimaryContact($cid)) {
+        // The contact being overwritten is a primary contact, so we don't allow for it to be overwritten.
+        return;
       }
+      $individualParams['contact_id'] = $cid;
     }
-    if (empty($individualParams['contact_id'])) {
+    else {
       $orgClause = ' r.contact_id_b = ' . $orgId;
       $orgClause = $isOrgOptional ? " ($orgClause OR r.contact_id_b IS NOT NULL) " : $orgClause;
       // Check for dupes.
-      $staffDetails = CRM_Core_DAO::executeQuery("SELECT r.contact_id_a, ca.first_name, ca.last_name
+      $existingStaff = CRM_Core_DAO::singleValueQuery("SELECT r.contact_id_a
          FROM civicrm_relationship r
          INNER JOIN civicrm_contact cb ON cb.id = r.contact_id_b
          LEFT JOIN civicrm_contact ca ON ca.id = r.contact_id_a
          WHERE $orgClause AND r.relationship_type_id = %2 AND r.is_active = 1
          AND ca.first_name = %3 AND ca.last_name = %4",
-        [2 => [$rel, "Integer"], 3 => [$individualParams['first_name'], 'String'], 4 => [$individualParams['last_name'], 'String']])->fetchAll()[0]; // We expect only a single contact
-      if (!empty($staffDetails)) {
+        [2 => [$rel, "Integer"], 3 => [$individualParams['first_name'], 'String'], 4 => [$individualParams['last_name'], 'String']]); // We expect only a single contact
+      if (!empty($existingStaff)) {
         // Dupe found
-        $individualParams["contact_id"] = $staffDetails['contact_id_a'];
+        if (E::checkPrimaryContact($existingStaff)) {
```

Comment:
This feels wrong to me, in that we have said we haven't been passed a contact_id so its a new contact be it regulated or ABA staff and we have now done a search to find a potential match and found it as the primary contact shouldn't we be just including the contact_id here?

Reply:
Again I don'think this has been addressed either @Edzelopez

Reply:
Addressed now.